platform/ios/src/MGLMapView.mm

```diff
@@ -4147,6 +4147,18 @@ - (MGLAnnotationTag)annotationTagAtPoint:(CGPoint)point persistingResults:(BOOL)
         }
         else
         {
+            // The first selection in the cycle should be the one nearest to the
+            // tap.
+            CLLocationCoordinate2D currentCoordinate = [self convertPoint:point toCoordinateFromView:self];
+            std::sort(nearbyAnnotations.begin(), nearbyAnnotations.end(), [&](const MGLAnnotationTag tagA, const MGLAnnotationTag tagB) {
+                CLLocationCoordinate2D coordinateA = [[self annotationWithTag:tagA] coordinate];
+                CLLocationCoordinate2D coordinateB = [[self annotationWithTag:tagB] coordinate];
+                CLLocationDegrees deltaA = hypot(coordinateA.latitude - currentCoordinate.latitude,
+                                                 coordinateA.longitude - currentCoordinate.longitude);
+                CLLocationDegrees deltaB = hypot(coordinateB.latitude - currentCoordinate.latitude,
+                                                 coordinateB.longitude - currentCoordinate.longitude);
+                return deltaA < deltaB;
+            });
             // Remember the nearby annotations for the next time this method is
```

Comment:
... this seems to be a duplicate of the code in the other branch (above) - so can we move this sort out and above the if? In doing so, we could remove the initial sort at line 4102.

/cc @1ec5

Reply:
Thanks for your review. You are right. The near annotations sort code was duplicated which should be called only one time.
As far as I know, `_annotationsNearbyLastTap` was used to step to the next annotation if there was an annotation selected and nearby annotations count greater than 1.